--- origsrc/findlib-1.2.1/src/findlib/findlib_config.mlp	2007-11-13 07:31:45.000000000 -0600
+++ src/findlib-1.2.1/src/findlib/findlib_config.mlp	2008-01-08 00:32:00.656250000 -0600
@@ -20,7 +20,7 @@
   match Sys.os_type with
       "Unix" -> ".so"
     | "Win32" -> ".dll"
-    | "Cygwin" -> ".dll"
+    | "Cygwin" -> ".so"
     | "MacOS" -> ""        (* don't know *)
     | _ -> failwith "Unknown Sys.os_type"
 ;;
--- origsrc/findlib-1.2.1/src/findlib/fl_split.ml	2007-11-13 07:31:45.000000000 -0600
+++ src/findlib-1.2.1/src/findlib/fl_split.ml	2008-01-08 00:34:28.750000000 -0600
@@ -72,7 +72,7 @@
 let path_separator =
   match Sys.os_type with
       "Unix"   -> ':'
-    | "Cygwin" -> ';'   (* You might want to change this *)
+    | "Cygwin" -> ':'   (* You might want to change this *)
     | "Win32"  -> ';'
     | "MacOS"  -> failwith "Findlib: I do not know what is the correct path separator for MacOS. If you can help me, write a mail to gerd@gerd-stolpmann.de"
     | _ -> failwith "Findlib: unknown operating system"
